### DISCUSSION TITLE
Suggestions for UI Improvements

### DISCUSSION THREAD

**Discussion by ghost** [fontforge/fontforge#4757] [Show and tell]
Hi there, my name is Adolfo. I'm a big fan of FontForge since I start using it during my college years to now. I made a new theme for it, called [zDarkTheme](https://github.com/ASIDEdesignlab/zDarkTheme). During its creation, I found few things that can't just improve its integration but also can help any other theme implementation in the future.

I'm pretty new to GitHub and its workflow. I hope it's ok I put it all together here.

# Suggestions for UI Improvements.

### Use the latest macOS DSK for native dark mode support.
Making a new compilation with the latest macOS SDK may provide native support for dark mode through the `NSRequiresAquaSystemAppearance` key. I'm using the [FontForge release 2020-11-07](https://fontforge.org/en-US/downloads/mac-dl/) on macOS Mojave 10.14.6 (18G9216). I tried the method [here](https://forums.macrumors.com/threads/forcing-third-party-applications-to-use-dark-mode.2134382/), but only the Terminal command allowed me to force dark mode. 

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-0.png" width="600">

### Unify tool palettes appearance and behavior.

The tool pallets on the BitmapView should have the same appearance and behavior as the ones on the CharView.

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-1.png" width="600">

### Correct some icons usages and drop using duplicate icons.

Make and share use of the icon files for the menus options `Set Color` and `Select by Color`

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-2.png" width="600">

Set a different icon file (hintsclearhints.png) for the menu option `Clear Hints`

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-3.png" width="600">

Set the same icon file (overlaprm.png) to the menu option `Remove Overlap` in MetricsView (this option is currently using a duplicated icon file rmoverlap.png).

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-4.png" width="600">

Below a list of files icons that seem not used:
|    changeweight.png    |  fileclose2.png  |    paletteselectedbg.png    |
|:----------------------:|:----------------:|:---------------------------:|
| elementtilepath.png    | fliphor.png      | palettespirodisabled.png    |
| elementtilepattern.png | flipvert.png     | palettespiroup-selected.png |
| exclude.png            | inline.png       | rotate180.png               |
| extendcondense.png     | intersection.png | rotateccw.png               |
| fflogo.png             | oblique.png      | rotatecw.png                |
| fflogo13.png           | outline.png      | shadow.png                  |
| ffsplash1.png          | text12210.png    | skew.png                    |
| ffsplash2.png          | wireframe.png    | splash2019.png              |
| ffsplash3.png          |                  | rmoverlap.png*              |

> `rmoverlap.png` is in use but is duplicate of `overlaprm.png`

### Apply the same icon for the same options on all views.

Apply icon file viewlayers.png to `Layers` menu option on MetricsView.

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-5.png" width="600">

Apply icon file editrmundoes.png to `Remove Undoes` on BitmapView.

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-6.png" width="600">

### Enable new interface icon customization.

Apply new custom icons for the pointer and point indicators on the info bar in the CharView and the BitmapView.

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-7.png" width="600">

Replace and add new custom icons for `Expand Stroke` dialog.

<img src="https://raw.githubusercontent.com/ASIDEdesignlab/zDarkTheme/main/Support/Images/zDarkTheme-Improvements-8.png" width="600">

# Final Thoughts

If you need the icon files I've suggested, they are all included in [zDarkThem](https://github.com/ASIDEdesignlab/zDarkTheme). I know that all these changes will require time and effort, but I think it's well worth it. No doubt to contact if I can help. Thanks for such a great app!

**Comment by adolfo-ovalles** [comment]
Project recovered here: https://github.com/adolfo-ovalles/zThemes

**Comment by iorsh** [comment]
The icons are very impressive! Quite by coincidence, I started porting the views, specifically Bitmap View to GTK3 and SVG icons.

<img width="682" height="690" alt="image" src="https://github.com/user-attachments/assets/c4eb0615-7a39-45dd-b0b4-8f5b59c3a3fb" />

I redrew a few toolbar icons in a scalable format, but your icons look much more consistent and professional, and I would be glad to adopt them. Do you have scalable sources for them? I was unable to retrieve any vector data from PSD files.

Eventually, there should be no need in separate set of icons for light and dark theme. GTK toolkit supports symbolic SVG (https://docs.gtk.org/gtk4/icon-format.html), where a monochrome SVG icon is colored in whatever color the theme defines, and thus fits it automatically.

For reference, my new SVG icons are available in https://github.com/iorsh/fontforge/tree/gtk_bitmap_view/fontforgeexe/pixmaps/tango/svg . They don't support the symbolic specification, but it shouldn't be too difficult to implement.

**Comment by adolfo-ovalles** [comment]
Hi, thank you so much, I really appreciate that!

Actually, I just checked, and I do have the AI file I used to create the pixel-perfect Photoshop versions. I'd be happy to share that in the repo.

One thing to note: I manually adjusted the PSD/PNG icons afterward for pixel-perfect rendering at 72dpi, so there may be minor differences between the vector shapes and the final bitmap versions. I'll export the icons as individual SVG files, named to match the existing PNG filenames, so they should be easy to map directly to your GTK3 icon set regardless of how you decide to organize them on your end.

About the symbolic SVG — as you described, I believe this is what Adobe app interfaces use as well. That's a game-changer; your approach using it is the right long-term direction.

Thank you again for your kind words and for considering my repo. I hope you can actually implement it as a definitive improvement to FontForge.

Give me a bit of time to review the file and export/organize the SVGs. I'll let you know once they're up in the repo.

**Comment by adolfo-ovalles** [comment]
Hi @iorsh ,

Just wanted to let you know I've uploaded the source files to the repo — you can find everything here:

https://github.com/adolfo-ovalles/zThemes/tree/main/Source

This includes the AI file along with the exported SVGs, named to match the existing PNG filenames as mentioned. Hope this helps with your work on the GTK3 port!

Looking forward to seeing how it comes together.

**Comment by iorsh** [comment]
Looks great, thanks a lot! I think we can actually make a new theme with these icons, so that the legacy UI uses them too. It will take some effort, but I would be able to start integrating it after the upcoming release (probably September-October 2026). Some previews would be available earlier though.

For now, I suggest you open a PR into FontForge, which would put the light icons only in `fontforgeexe/pixmaps/<new_theme_name>/*.svg` (the dark icons will be derived automatically as we discussed). Please suggest the new name as you wish.

**Comment by adolfo-ovalles** [comment]
That's fantastic, thank you! I'd love to see the icons integrated into the legacy UI as well.

I'll go ahead and prepare a PR with the light icons in `fontforgeexe/pixmaps/<theme_name>/*.svg` as you suggested, using the existing **`Tango`** theme files as reference for naming.

For the name, I was thinking of keeping **`ztheme`**, but I'm open to whatever fits best with FontForge's existing naming conventions.

Really excited to see this move forward. I'll keep you posted.

**Reply by iorsh** [reply]
Thanks, "ztheme" is great, we don't really have any specific conventions. The current theme is "tango", and its predecessor was "2012".

**Comment by adolfo-ovalles** [comment]
Hi @iorsh,

Just wanted to let you know I've opened a draft PR with the ztheme SVG icon set — you can find it here: https://github.com/fontforge/fontforge/pull/5891

The PR's README covers naming details, a few open questions (icon consolidation, fill-color convention for symbolic theming), and known placeholders.

This is my first time contributing directly to a project like FontForge, so I really appreciate your patience and guidance along the way. Let me know what you think whenever you get a chance!
